workspace_tools/host_tests/serial_nc_tx_auto.py

```diff
@@ -0,0 +1,59 @@
+"""
+mbed SDK
+Copyright (c) 2011-2013 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+import uuid
+import time
+import string
+from sys import stdout
+
+class SerialNCTXTest():
+
+    def test(self, selftest):
+        selftest.mbed.flush();
+        selftest.mbed.serial_write("S");
+        
+        strip_chars = string.whitespace + "\0"
+
+        out_str = selftest.mbed.serial_readline()
+        selftest.notify("HOST: " + out_str)
+
+        if out_str == None:
+            selftest.notify("HOST: No output detected")
+            return selftest.RESULT_IO_SERIAL
+
+        out_str_stripped = out_str.strip(strip_chars)
+
+        if out_str_stripped != "TX OK - Expected":
+            selftest.notify("HOST: Unexpected output. Expected 'TX OK - Expected' but received '%s'" % out_str_stripped)
+            return selftest.RESULT_FAILURE
+
+        out_str = selftest.mbed.serial_readline()
+
+        # If no characters received, pass the test
+        if out_str == "" or out_str == None:
```

Comment:
Regarding:
```python
if out_str == ""
```
For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
```
if not out_str or out_str is None:
```